apps/src/sites/studio/pages/blocks/edit.js

```diff
@@ -70,36 +72,65 @@ function initializeEditPage(defaultSprites) {
 
   if (blocks) {
     updateBlockPreview();
+    validateBlockConfig();
   }
+  setSubmitButtonState();
 
   $('.alert.alert-success')
     .delay(5000)
     .fadeOut(1000);
 }
 
 function onUpdateLinting(_, errors) {
-  const submitButton = document.querySelector('#block_submit');
   if (errors.length) {
-    submitButton.setAttribute('disabled', 'disabled');
+    hasLintingErrors = true;
   } else {
-    submitButton.removeAttribute('disabled');
+    hasLintingErrors = false;
   }
+  setSubmitButtonState();
 }
 
+const setSubmitButtonState = () => {
+  const disabled = hasLintingErrors || !isValidBlockConfig;
+  ['#block_clone', '#block_submit'].forEach(buttonId => {
+    const button = document.querySelector(buttonId);
+    if (disabled) {
+      button.setAttribute('disabled', 'disabled');
+    } else {
+      button.removeAttribute('disabled');
+    }
+  });
+};
+
 function validateBlockConfig(editor) {
   try {
     if (editor) {
       JSON.parse(editor.getValue());
     }
     updateBlockPreview();
+    validateBlockRenders();
+    isValidBlockConfig = true;
     validationDiv.text('Config and helper code appear valid.');
     validationDiv.css('color', VALID_COLOR);
   } catch (err) {
+    isValidBlockConfig = false;
     validationDiv.text(err.toString());
     validationDiv.css('color', INVALID_COLOR);
   }
 }
 
+// Validation only relevant for blocks rendered via google blockly
+// "unknown block" is only implemented in google blockly
+function validateBlockRenders() {
+  if (
+    Blockly.mainBlockSpace
+      .getAllBlocks()
+      .some(block => block.getFieldValue('NAME')?.includes('unknown block'))
```

Comment:
Just of curiosity, is there a way to cache this calculation so that we don't have to call `Blockly` on every function call? Maybe it's possible to pass all blocks as an argument to that function? (I don't know much about this part of project, so that's are just my first sight suggestion)